app/hydrosheds_app.cc

```diff
@@ -1,35 +1,275 @@
-#include "hydrosheds/hydrosheds.hh"
+#include "../include/hydrosheds/hydrosheds.hh"
 
 #include "ogrsf_frmts.h"
 
-#include<iostream>
+#include <iostream>
+#include <cmath>
+#include <iomanip>
 
+using namespace hydrosheds;
 
-int main(int argc, char** argv)
+// static layer declaration
+OGRLayer* RiverSegment::layer;
+
+/* -- CLASS HYDROSHEDS DATA SET -- */
+HydroshedsDataSet::HydroshedsDataSet(const std::string& path)
```

Comment:
The implementation of the interface should be in `src/hydrosheds.cc`, not in the app. Otherwise, it will not properly build into a library. The app is really just something for us to toy around with - a user would typically link against the library and use it from their C++ code.